ipsource.go

```diff
@@ -38,7 +38,14 @@ func init() {
 
 // IPSource is a type that can get IP addresses.
 type IPSource interface {
-	GetIPs(context.Context, IPVersions) ([]net.IP, error)
```

Comment:
This is a breaking change for third-party IPSource plugins!!!

If we want to carry IP range settings into IPSource plugins (as opposed to lowering IP range settings into each plugin that wants to use it) this will be a binary-incompatible change no matter what.

It could be a source-compatible change by smuggling in IP range settings into the existing IPVersions type, but this would be pretty ugly and confusing.

Reply:
Alternatively, can just include this config directly on each ip_source plugin which wishes to use it